### PR TITLE
test/key_gen: DKG procedure failed

### PR DESCRIPTION
**Pull request by maqi**

### CODE REVIEW

File: src/key_gen/tests.rs [hunk 1]
```diff
@@ -154,21 +157,74 @@ fn having_one_non_responsive_node() {
...
+                    assert!(!key_gen.pub_keys().contains(&peer_ids[*idx as usize]))
+                });
             } else {
                 assert!(key_gen.generate_keys().is_none());
             }
         });
 }
 
+#[test]
+fn having_more_non_responsive_nodes() {
```
Comment (madadam) [inline src/key_gen/tests.rs]: Nice, but should we have two tests: one where the number of non-responsives is the maximum where completion is still possible (I assume `n - (t + 1)` ?) and one where it is the minimum where completion is blocked (so `n - t` ?).

**Review by madadam — COMMENTED** [review]
Looks good, just an idea to possibly expand the tests a bit.

File: src/key_gen/tests.rs (hunk 1 above)
Comment (maqi) [reply src/key_gen/tests.rs]: given the current setup of n=7 and m=5, the previous test of having_one_non_responsive_node is actuall testing `n - (t + 1)`. I will update the name of that test and the setup part to reflect this.
For this test, I will change it to `n-t` to cover the `minimum of causing blocking`

**Review by Yoga07 — APPROVED** [review]
LGTM! :+1:

**Review by madadam — APPROVED** [review]

**Pull request merged**